tiledb/sm/compressors/delta_compressor.cc

```diff
@@ -121,12 +129,11 @@ void Delta::compress(
       break;
     case Datatype::FLOAT32:
     case Datatype::FLOAT64:
-      throw StatusException(
-          Status_CompressionError("Cannot compress tile with Delta; Float "
-                                  "datatypes are not supported"));
+      throw DeltaCompressorException(
+          "Compression is not yet supported for float datatypes.");
     default:
-      throw StatusException(Status_CompressionError(
-          "Cannot compress tile with Delta; Unsupported datatype"));
+      throw DeltaCompressorException(
+          "Compression failed; Unsupported datatype");
```

Comment:
Nit: Use `datatype_str` to say which datatype is unsupported.

Reply:
`datatype_str` will return an empty string here because each `Datatype` case is covered above.

Reply:
Should it be "invalid" datatype and the integer value instead then? Probably not too important since (in theory) we should never hit this error.